sdk/python/kfp/_local_client.py

```diff
@@ -0,0 +1,420 @@
+    ):
+        op_dependency = pipeline.ops[op_name]
+        if output_name is None and len(op_dependency.file_outputs) == 1:
+            output_name = next(iter(op_dependency.file_outputs.keys()))
+        output_file = op_dependency.file_outputs[output_name]
+        altered_output_file = self._alter_output_file_path(
+            run_name, op_name, output_file
+        )
+        return altered_output_file
+
+    def _op_cmd_locally(
+        self,
+        run_name: str,
+        pipeline: kfp.dsl.Pipeline,
+        op: kfp.dsl.ContainerOp,
+        stack: Dict[str, Any],
+    ):
+        cmd = op.command + op.arguments
+
+        # In debug mode, for `python -c cmd` format command, pydev will insert code before
+        # `cmd`, but there is no newline at the end of the inserted code, which will cause
+        # syntax error, so we add newline before `cmd`.
+        for i in range(len(cmd)):
+            if cmd[i] == "-c":
+                cmd[i + 1] = "\n" + cmd[i + 1]
+
+        _arg_pairs = self._pair_arguments(cmd)
+        _fixed_arg_value = {}
+        for arg_name, arg_value in _arg_pairs:
+            if arg_value in stack:  # Argument is LoopArguments item
+                arg_value = str(stack[arg_value])
+            elif arg_value in op.file_outputs.values():  # Argument is output file
+                output_name = next(
+                    filter(lambda item: item[1] == arg_value, op.file_outputs.items())
+                )[0]
+                output_param = op.outputs[output_name]
+                output_file = arg_value
+                output_file = self._alter_output_file_path(
+                    run_name, output_param.op_name, output_file
+                )
+                arg_value = output_file
+            elif (
+                arg_value in op.input_artifact_paths.values()
+            ):  # Argument is input artifact file
+                input_name = next(
+                    filter(
+                        lambda item: item[1] == arg_value,
+                        op.input_artifact_paths.items(),
+                    )
+                )[0]
+                input_param_pattern = op.artifact_arguments[input_name]
+                pipeline_param = self._extract_pipeline_param(input_param_pattern)
+                input_file = self.get_output_file_path(
+                    run_name, pipeline, pipeline_param.op_name, pipeline_param.name
+                )
+
+                arg_value = input_file
+            _fixed_arg_value[arg_name] = arg_value
+
+        cmd = self._replace_file_args(cmd, _fixed_arg_value)
+        return cmd
+
+    def _op_cmd_on_docker(
+        self,
+        run_name: str,
+        pipeline: kfp.dsl.Pipeline,
+        op: kfp.dsl.ContainerOp,
+        stack: Dict[str, Any],
+    ):
+        cmd = self._op_cmd_locally(run_name, pipeline, op, stack)
+        cmd[
+            2
+        ] = 'program_path=$(mktemp)\n echo -n "$0" > "$program_path"\n python3 -u "$program_path" $@ '
```

Comment:
`echo -n` needs to be replaced with `printf "%s"`

Reply:
Sorry, forgot to remove code used during development, `cmd[2] = ..` is useless, and has been removed